### PR TITLE
Added Note Type & Custom Type definitions

### LINKED ISSUE #7961: New Topic - How to create custom field types for system fields (admin configuration)
# New topic request

## Description

/guides/v2.4/config-guide/prod/config-reference-systemxml.html - will soon have an updated section as to the different field types available for system fields. However, it is possible to create and use custom field types. A new topic to describe how to use this feature, with examples, would be desirable. With a link to it from the original topic.

## Content checklist

-  [ ] The topic provides an explanation of how custom field types work.
-  [ ] The topic provides steps for creating and using custom field types.
-  [ ] The topic contains code samples that shows how these are created.
-  [ ] Must have a link from the original topic

### PR DESCRIPTION
**Pull request by BarnyShergold**
## Purpose of this pull request

This pull request (PR) fixes #7961 & adds 'note' type

## Affected DevDocs pages
https://devdocs.magento.com/guides/v2.3/config-guide/prod/config-reference-systemxml.html
https://devdocs.magento.com/guides/v2.4/config-guide/prod/config-reference-systemxml.html

### CODE REVIEW

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 1]
```diff
@@ -249,6 +249,22 @@ A `<field>`-Tag can have the following values for the `type=""` attribute:
 | `time`          | Control to set time using three dropdowns–Hour, minute and second.                                                                                                                                         |
 | `allowspecific` | A multiselect list of specific countries. Requires a `source_model` such as `Magento\Shipping\Model\Config\Source\Allspecificcountries`                                                                               |
 | `image`         | Allows an image to be uploaded.                                                                                                                                                                                                                                                                                                                                   |
+| `note`          | Allows an informational note to be added to the page. This type requires a `frontend_model` to render the note. |
+
+It is also possible to create a custom field type. This is often done when a special button, with an action, is required. To do this requires two main elements:
+
+-  Creating a Block in the `adminhtml` area
```
Comment (dobooth) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:256]: lowercase 'block'. Are there 3 quotes in `type=`?

File: src/guides/v2.3/config-guide/prod/config-reference-systemxml.md [hunk 2]
```diff
@@ -249,6 +249,22 @@ A `<field>`-Tag can have the following values for the `type=""` attribute:
...
+-  Setting the `type="""` to the path to this block
+
+The block itself requires, at a minimum, a `__contstuct` method and a `getElementalHtml()` method. The [Magento_OfflineShipping](https://github.com/magento/magento2/tree/2.4-develop/app/code/Magento/OfflineShipping) is a simple example of a custom type.
```
Comment (dobooth) [inline src/guides/v2.3/config-guide/prod/config-reference-systemxml.md:259]: `__construct`
The github link will fail after 2.4-develop is merged. There is a var for the repo.

**Review by dobooth — CHANGES_REQUESTED** [review]

**Comment by dobooth** [comment]
@BarnyShergold

**Review by dobooth — APPROVED** [review]

**Review by rogyar — APPROVED** [review]

**Comment by BarnyShergold** [comment]
@dobooth - Can this not be completed now? Approvals were done a week ago 😄

**Comment by keharper** [comment]
running tests

**Pull request merged**